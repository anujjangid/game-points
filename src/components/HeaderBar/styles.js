import styled from 'styled-components';

export const Header = styled.header`
  color: white;
  background: #8acdd2;
  box-sizing: border-box;
  h4 {
    @media only screen and (min-width: 768px) {
      font-size: 24px;
    }
    font-size: 13px;
    font-weight: 400;
    padding: 20px;
    text-transform: uppercase;
  }
`;
