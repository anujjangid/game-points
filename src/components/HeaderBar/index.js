import React from 'react';
import PropTypes from 'prop-types';

import { Header } from './styles';

const HeaderBar = ({ title }) => {
  return (
    <React.Fragment>
      <Header>
        <h4>{title}</h4>
      </Header>
    </React.Fragment>
  );
};

export default HeaderBar;

HeaderBar.propTypes = {
  title: PropTypes.string
};

HeaderBar.defaultProps = {
  title: 'Test'
};
