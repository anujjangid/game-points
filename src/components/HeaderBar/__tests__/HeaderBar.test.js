import React from 'react';

import { create } from 'react-test-renderer';

import HeaderBar from '../index';

describe('Test suite for HeaderBar Component', () => {
  // create a UI snapshot to indentify UI changes or design regressions
  it('render correctly', () => {
    const component = create(<HeaderBar />);
    expect(component.toJSON()).toMatchSnapshot();
    const rootInstance = component.root;

    const Heading = rootInstance.findByType('h4');
    expect(Heading.props.children).toBe('Test');
  });
});
