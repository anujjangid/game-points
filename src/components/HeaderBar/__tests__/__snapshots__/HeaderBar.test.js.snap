// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for HeaderBar Component render correctly 1`] = `
<header
  className="sc-bdVaJa eQZKMo"
>
  <h4>
    Test
  </h4>
</header>
`;
